test/IIS.FunctionalTests/Inprocess/StdOutRedirectionTests.cs

```diff
@@ -0,0 +1,218 @@
+                var count = 0;
+                foreach (var write in TestSink.Writes)
+                {
+                    if (write.Message.Contains(
+                        "The specified framework 'Microsoft.NETCore.App', version '2.9.9' was not found."))
+                    {
+                        count++;
+                    }
+                }
+
+                if (deploymentParameters.ServerType == ServerType.IISExpress)
+                {
+                    Assert.Equal(2, count);
+                }
+                else
+                {
+                    Assert.Equal(1, count);
+                }
+            }
+            finally
+            {
+                Directory.Delete(pathToLogs);
+            }
+        }
+
+        [ConditionalFact]
+        [SkipIfDebug]
+        public async Task EnableCoreHostTraceLogging_TwoLogFilesCreated()
+        {
+            var pathToLogs = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+
+            try
+            {
+                var deploymentParameters =
+                    _fixture.GetBaseDeploymentParameters(_fixture.StartupExceptionWebsite, publish: true);
+                deploymentParameters.EnvironmentVariables["COREHOST_TRACE"] = "1";
+                deploymentParameters.GracefulShutdown = true;
+
+                deploymentParameters.EnableLogging(pathToLogs);
+
+                var deploymentResult = await DeployAsync(deploymentParameters);
+
+                var response = await deploymentResult.HttpClient.GetAsync("/");
+                Assert.False(response.IsSuccessStatusCode);
+
+                StopServer();
+
+                var filesInDirectory = Directory.GetFiles(pathToLogs);
+                Assert.Equal(2, filesInDirectory.Length);
+            }
+            finally
+            {
+                Directory.Delete(pathToLogs);
+            }
+        }
+
+        [ConditionalTheory]
+        [SkipIfDebug]
+        [InlineData("CheckLargeStdErrWrites")]
+        [InlineData("CheckLargeStdOutWrites")]
+        [InlineData("CheckOversizedStdErrWrites")]
+        [InlineData("CheckOversizedStdOutWrites")]
+        public async Task EnableCoreHostTraceLogging_PipeRestoreCorrectly(string path)
+        {
+            var deploymentParameters = _fixture.GetBaseDeploymentParameters(_fixture.StartupExceptionWebsite, publish: true);
+            deploymentParameters.EnvironmentVariables["COREHOST_TRACE"] = "1";
+            deploymentParameters.WebConfigBasedEnvironmentVariables["ASPNETCORE_INPROCESS_STARTUP_VALUE"] = path;
+            deploymentParameters.GracefulShutdown = true;
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
+
+            var response = await deploymentResult.HttpClient.GetAsync("/");
+
+            Assert.False(response.IsSuccessStatusCode);
+
+            StopServer();
+
+            Assert.Contains(TestSink.Writes, context => context.Message.Contains("Invoked hostfxr"));
```

Comment:
Why are we asserting this message? and not writes coming from managed?

Reply:
I'll rename the file. These tests are only for getting native logs. StartupExceptionTests contains the other tests as they can run on both IIS and IISExpress.